scripts/heat/instance_pairs.yaml

```diff
@@ -0,0 +1,186 @@
+    type: string
+    description: End of private network IP address allocation pool
+    default: 172.20.1.99
+
+resources:
+
+  private_net_east:
+    type: OS::Neutron::Net
+    properties:
+        name: { get_param: east_net_name }
+
+  private_subnet_east:
+    type: OS::Neutron::Subnet
+    properties:
+      network_id: { get_resource: private_net_east }
+      cidr: { get_param: east_net_cidr }
+      allocation_pools:
+        - start: { get_param: east_net_pool_start }
+          end: { get_param: east_net_pool_end }
+
+  private_net_west:
+    type: OS::Neutron::Net
+    properties:
+        name: { get_param: west_net_name }
+
+  private_subnet_west:
+    type: OS::Neutron::Subnet
+    properties:
+      network_id: { get_resource: private_net_west }
+      cidr: { get_param: west_net_cidr }
+      allocation_pools:
+        - start: { get_param: west_net_pool_start }
+          end: { get_param: west_net_pool_end }
+
+  router:
+    type: OS::Neutron::Router
+    properties:
+      external_gateway_info:
+        network: { get_param: public_net }
+
+  router_interface_east:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: router }
+      subnet_id: { get_resource: private_subnet_east }
+
+  router_interface_west:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: router }
+      subnet_id: { get_resource: private_subnet_west }
+
+  security_group:
+    type: OS::Neutron::SecurityGroup
+    properties:
+      rules:
+        - direction: ingress
+          ethertype: IPv4
+
+  east_port:
+    type: OS::Neutron::Port
+    properties:
+      security_groups:
+        - { get_resource: security_group }
+      network_id: { get_resource: private_net_east }
+      fixed_ips:
+        - subnet_id: { get_resource: private_subnet_east }
+
+  east_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: { get_param: public_net }
+      port_id: { get_resource: east_port }
+
+  west_port:
+    type: OS::Neutron::Port
+    properties:
+      security_groups:
+        - { get_resource: security_group }
+      network_id: { get_resource: private_net_west }
+      fixed_ips:
+        - subnet_id: { get_resource: private_subnet_west }
+
+  west_floating_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: { get_param: public_net }
+      port_id: { get_resource: west_port }
+
+  east:
+    type: OS::Nova::Server
+    properties:
+      name: { get_param: east_name }
+      image: { get_param: image }
+      flavor: { get_param: flavor }
+      key_name: testkey
+      networks:
+        - port: { get_resource: east_port }
+      user_data_format: RAW
+      user_data:
+        str_replace:
+          template: { get_file: ping_template.sh }
+          params:
+            "@outside_ip@": { get_param: outside_ip }
+            "@neighbour_ip@": { get_attr: [ west_floating_ip, floating_ip_address ] }
```

Comment:
Strictly speaking, this is not simply east-west traffic. Rather, it's east-north-north-west, since traffic goes from one private network to the floating IP network, out from one floating IP, back in through another floating IP and from there to the other private network. Is that the intention? For otherwise one could use the private IP addresses (the router is aware of both networks after all).